pkg/tfgen/main.go

```diff
@@ -33,14 +33,38 @@ import (
 
 // Main executes the TFGen process for the given package pkg and provider prov.
 func Main(pkg string, version string, prov tfbridge.ProviderInfo) {
-	if err := newTFGenCmd(pkg, version, prov).Execute(); err != nil {
+	MainWithCustomGenerate(pkg, version, prov, func(opts GeneratorOptions) error {
+
+		// Create a generator with the specified settings.
+		g, err := NewGenerator(opts)
+		if err != nil {
+			return err
+		}
+
+		// Let's generate some code!
+		err = g.Generate()
+		if err != nil {
+			return err
+		}
+
+		return err
+	})
+}
+
+// Like Main but allows to customize the generation logic past the parsing of cmd-line arguments.
```

Comment:
@guineveresaenger and @viveklak  I had questions about this in a few places, answering here. This annoying looking logic supports a divergent behavior between legacy bridged providers and TFPF providers around generating the renames.json file. 

So that:

- legacy bridged providers continue using Main that does not generate renames.json
- whereas TFPF bridged providers use a MainWithCustomGenerate that goes ahead and generates the renames.json

SO the good news is that accepting this PR does not generate any downstream work in the bridged provider boilerplates, but the bad news perhaps is that this does not look particularly elegant or clear; if you have a better idea of where to make the option of generating renames configurable LMK I can clean this up in a follow-up PR.